p2p/conn/secret_connection.go

```diff
@@ -162,8 +164,10 @@ func MakeSecretConnection(conn io.ReadWriteCloser, locPrivKey crypto.PrivKey) (*
 	}
 
 	remPubKey, remSignature := authSigMsg.Key, authSigMsg.Sig
-	if _, ok := remPubKey.(ed25519.PubKey); !ok {
-		return nil, fmt.Errorf("expected ed25519 pubkey, got %T", remPubKey)
+	switch remPubKey.(type) {
+	case ed25519.PubKey, secp256k1.PubKey, secp256k1_eth.PubKey:
+	default:
+		return nil, fmt.Errorf("expected ed25519/secp256k1/secp256k1_eth pubkey, got %T", remPubKey)
```

Comment:
This allows testing end-to-end

Reply:
why are you changing the secret connection key type? this doesnt effect what key the validator is signing with. IMO this shouldnt be touched since its already very fragile

Reply:
I had to change it when running `e2e` tests, otherwise it wasn't working, so that was my immediate fix without much thought (this is still WIP). But following your comment, I'll take a closer look at this.

Reply:
Fixed